isis/src/base/objs/Blob/Blob.cpp

```diff
@@ -368,6 +368,19 @@ namespace Isis {
     }
   }
 
+  void Blob::setData(const char *buffer, int nbytes) {
+    p_nbytes = nbytes;
+
+    if (p_buffer != NULL) {
+      delete [] p_buffer;
+    }
+    p_buffer = new char[p_nbytes];
+
+    for (int i = 0; i < p_nbytes; i++) {
+      p_buffer[i] = buffer[i];
+    }
+  }
```

Comment:
It's possible this function should just pass ownership of the buffer to the Blob to avoid copying potentially quite large chunks of data.